pkg/nuclide-rust/lib/BuckIntegration.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the LICENSE file in
+ * the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+import type {TaskInfo} from '../../nuclide-buck/lib/types';
+import type {
+  AtomLanguageService,
+  LanguageService,
+} from '../../nuclide-language-service';
+
+import invariant from 'invariant';
+import {getLogger} from 'log4js';
+import fsPromise from 'nuclide-commons/fsPromise';
+import nuclideUri from 'nuclide-commons/nuclideUri';
+import {getRustInputs, getSaveAnalysisTargets, normalizeNameForBuckQuery} from './BuckUtils';
+
+import * as BuckService from '../../nuclide-buck-rpc';
+
+const logger = getLogger('nuclide-rust');
+
+export async function updateRlsBuildForTask(
+  task: TaskInfo,
+  service: AtomLanguageService<LanguageService>,
+) {
+  const buildTarget = normalizeNameForBuckQuery(task.buildTarget);
+
+  // TODO: Filter by known Rust build targets
+  const files = await getRustInputs(task.buckRoot, buildTarget);
```

Comment:
This still requires explicit filtering for `rust_*` target kinds.

Reply:
please do this filtering to avoid calling the expensive `buck query` unless we're sure this is a rust target & this call will have value.

Reply:
Done in https://github.com/facebook/nuclide/pull/1606/commits/8d4772f9ecda99b132d0f4b8eced16b9124455af. Forgot to remove the comment, sorry!